jina/types/document/uid.py

```diff
@@ -88,3 +91,35 @@ def is_valid_id(value: str) -> bool:
         and "A"–"F" (or alternatively "a"–"f"). \
         - it has 16 chars described above.')
     return True
+
+
+class UniqueId(str):
+    def __new__(cls, seq):
+        if isinstance(seq, (int, np.integer)):
+            seq = hash2id(int(seq))
+        elif isinstance(seq, bytes):
+            seq = bytes2id(seq)
+        elif seq == '':
+            pass
+        elif isinstance(seq, str) and is_valid_id(seq):
+            seq = seq
+        elif seq is not None:
+            raise BadDocID(f'{typename(seq)}: {seq} is not a valid id')
+
+        return str.__new__(cls, seq)
+
+    @property
+    def to_hash(self):
+        """The document id in the integer form of bytes, as 8 bytes map to int64.
+        This is useful when sometimes you want to use key along with other numeric values together in one ndarray,
+        such as ranker and Numpyindexer
+        """
+        return id2hash(self)
```

Comment:
can we do something like this? `to_hash` sounds like a behaviour, not a property
```python
def __hash__(self):
    return id2hash(self)
```

Reply:
`hash` is built_in tho, can change it to `as_hash`

Reply:
change to `hash_value`

Reply:
@hanxiao no, the point is. `hash_value` is an "action"/"behaviour", not a "property".  the name `to_hash` or `hash_value` both sounds good to me. while it's a method performing some action on an instance, can we rewrite the default magic method? (i.e. `__hash__`)

Reply:
it is a computational property, so a property

Reply:
i will override __bytes__ and __hash__